x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/cache/blob/SearchableSnapshotsBlobStoreCacheMaintenanceIntegTests.java

```diff
@@ -311,6 +315,46 @@ public void testPeriodicMaintenance() throws Exception {
         }
     }
 
+    public void testCleanUpMigratedSystemIndexAfterIndicesAreDeleted() throws Exception {
+        final String repositoryName = "repository";
+        createRepository(repositoryName, FsRepository.TYPE);
+
+        final Map<String, Tuple<Settings, Long>> mountedIndices = mountRandomIndicesWithCache(repositoryName, 3, 10);
+        ensureYellow(SNAPSHOT_BLOB_CACHE_INDEX);
+        refreshSystemIndex(true);
+
+        final long numberOfEntriesInCache = numberOfEntriesInCache();
+        logger.info("--> found [{}] entries in snapshot blob cache", numberOfEntriesInCache);
+        assertThat(numberOfEntriesInCache, equalTo(mountedIndices.values().stream().mapToLong(Tuple::v2).sum()));
+
+        migrateTheSystemIndex();
+
+        logger.info("--> deleting indices, maintenance service should clean up snapshot blob cache index");
+        assertAcked(indicesAdmin().prepareDelete("mounted-*"));
+        assertBusy(() -> {
+            refreshSystemIndex(true);
+            assertHitCount(systemClient().prepareSearch(SNAPSHOT_BLOB_CACHE_INDEX).setSize(0), 0L);
+        });
+    }
+
+    /**
+     *  Mimics migration of the {@link SearchableSnapshots#SNAPSHOT_BLOB_CACHE_INDEX} as done in
+     *  {@link org.elasticsearch.upgrades.SystemIndexMigrator}, where the index is re-indexed, and replaced by an alias.
+     */
+    private void migrateTheSystemIndex() {
```

Comment:
I did have a quick look at `SystemIndexMigrator` hoping I could trigger this migration step by creating/running such a migrator. But that doesn't seem straight-forward and it would still be mimicking a proper upgrade test which is what is missing here. Considering the lack of automated upgrade tests that include the migration steps is a bigger issue and not addressed currently, I kept the test here short just trying to reindex/alias/remove, which is what the migrator is doing.

I have also manually verified this by creating/mounting a searchable snapshot index in 7.x, and upgrading to 8.x (with migration) and then 9.x and removing the searchable snapshot index in 9.x which cleans up the migrated snapshot blob cache index.

Reply:
So we have no actual upgrade tests that run system index migrations? That indeed sounds problematic and something we should ensure happens (but fine outside this PR, for another team too I think).

Reply:
My understanding is that we don't have that. which is why we resorted to manual testing in the first place. I can follow up and ask around, and create a follow up issue or PR if it turns out to be possible.

Reply:
In #121517 Christoph reused the upgrade tests we made for N-2 support in order to test the upgrade of the `async-search` system index.

Reply:
Thanks. I'll look into that. back then I asked core/infra about v7->v8->v9 and they mentioned there is currently no option. But this could be partially useful.